src/lib.rs

```diff
@@ -121,23 +124,35 @@ impl Build {
         false
     }
 
+    /// Exits the process on failure. Use `try_build` to handle the error.
     pub fn build(&mut self) -> Artifacts {
-        let target = &self.target.as_ref().expect("TARGET dir not set")[..];
-        let host = &self.host.as_ref().expect("HOST dir not set")[..];
-        let out_dir = self.out_dir.as_ref().expect("OUT_DIR not set");
+        match self.try_build() {
+            Ok(a) => a,
+            Err(e) => {
+                println!("cargo:warning=openssl-src: failed to build OpenSSL from source");
+                eprintln!("\n\n\n{e}\n\n\n");
+                std::process::exit(101); // same as a panic
```

Comment:
It should probably use 1 as exit code given that it isn't a bug in the build script that caused a crash.

Reply:
ok